junit-runtime/src/main/scala/org/scalajs/junit/JUnitMasterRunner.scala

```diff
@@ -0,0 +1,82 @@
+package org.scalajs.junit
+
+import com.novocode.junit.RunSettings
+import sbt.testing._
+import java.util.concurrent.atomic.AtomicInteger
+
+final class JUnitMasterRunner(
+    args: Array[String],
+    remoteArgs: Array[String],
+    testClassLoader: ClassLoader,
+    runSettings: RunSettings)
+    extends JUnitBaseRunner(args, remoteArgs, testClassLoader, runSettings) {
+
+  private[this] var registeredCount = 0
+
+  private[this] var doneCount = 0
+  private[this] var passedCount = 0
+  private[this] var failedCount = 0
+  private[this] var ignoredCount = 0
+  private[this] var skippedCount = 0
+  private[this] var totalCount = 0
+
+  private[this] var slaveCount = 0
+
+  def tasks(taskDefs: Array[TaskDef]): Array[Task] = {
+    registeredCount += taskDefs.length
+    taskDefs.map(newTask)
+  }
+
+  def done(): String = {
+    val slaves = slaveCount
+    val registered = registeredCount
+    val done = doneCount
+
+    if (slaves > 0)
+      throw new IllegalStateException(s"There are still $slaves slaves running")
+
+    if (registered != done) {
+      val msg = s"$registered task(s) were registered, $done were executed"
+      throw new IllegalStateException(msg)
+    } else {
+      val skipped = skippedCount
+      s"""Passed: Total $totalCount,
+         |Errors $failedCount,
+         |Passed $passedCount,
+         |${if (skipped != 0) s"Skipped $skipped" else ""}
+         |""".stripMargin.replace('\n', ' ')
+    }
+  }
+
+  private[junit] def taskDone(): Unit = doneCount += 1
+  private[junit] def taskPassed(): Unit = passedCount += 1
+  private[junit] def taskFailed(): Unit = failedCount += 1
+  private[junit] def taskIgnored(): Unit = ignoredCount += 1
+  private[junit] def taskSkipped(): Unit = skippedCount += 1
+  private[junit] def taskRegisterTotal(): Unit = totalCount += 1
+
+  private[junit] def taskPassedCount(): Int = passedCount
+  private[junit] def taskFailedCount(): Int = failedCount
+  private[junit] def taskIgnoredCount(): Int = ignoredCount
+  private[junit] def taskSkippedCount(): Int = skippedCount
+  private[junit] def taskTotalCount(): Int = totalCount
+
+  def receiveMessage(msg: String): Option[String] = msg(0) match {
+    case 's' =>
+      slaveCount += 1
+      Some("Hello")
```

Comment:
I still think maintaining `slaveCount` and handling this `'s'` message is not necessary.